### ISSUE TITLE
Speedup and reduced memory usage for debevec hdr creator

### ISSUE THREAD

**Issue by heckflosse** [LuminanceHDR/LuminanceHDR#94]
I created a [new branch](https://github.com/heckflosse/LuminanceHDR/tree/debevec_speedup) in my fork with a speedup and reduced memory usage for debevec

I used a single 24 Mpixel file for my tests. Here are the timings:

```
                       before                    after
debevec               9740 ms                   820 ms
```

The speedup depends on the number of input images you use to create a hdr and the number of cores.

The memory usage is reduced by
```
min(i, c) * 24 * width * height byte
```
where `i` is the number of input images and `c` is the number of cores.
Means, creating an HDR from 4 36 Mpixel input images on a 4-core machine needs ~3.4 GB less memory than before.

@Benitoite I hope the nested omp stuff works on mac too. If not I will make that part conditional.

**Comment by Benitoite** [comment]
It actually seems to work well. To get it to compile, though, I have to say `std::isnormal(` and `!std::isnormal(` because the system thinks `isnormal(` by itself is ambiguous.

**Comment by heckflosse** [comment]
@Benitoite I pushed a fix to my branch

**Comment by Benitoite** [comment]
Mac build: https://filebin.net/rycs3cdbfe5gnxrz/Luminance_HDR_2.5.2-SNAPSHOT-Test1h.dmg

**Comment by heckflosse** [comment]
Fixed with 0b394f7
